tests/suites/storage/charm_storage.sh

```diff
@@ -18,105 +16,110 @@ run_charm_storage() {
+	juju storage --format json | jq '.storage | keys | .[]' | check "data/0"
 	# assert the unit attachment name
-	juju list-storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
+	juju storage --format json | jq '.storage[] | .attachments | .units | keys | .[]' | check "dummy-storage-fs/0"
 	# assert the attached unit state
-	juju list-storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
+	juju storage --format json | jq '.storage[] | .attachments | .units[] | .life' | check "alive"
 	echo "Filesystem rootfs PASSED"
 	# remove the application
 	juju remove-application dummy-storage-fs
 	wait_for "{}" ".applications"
+
 	# Assess charm storage with the filesystem storage provider
 	echo "Assessing block loop disk 1"
 	juju deploy ./testcharms/charms/dummy-storage-lp --series="jammy" --storage disks=loop,1G
-	wait_for "dummy-storage-lp" ".applications"
+	wait_for "dummy-storage-lp" "$(idle_condition "dummy-storage-lp" 0 0)"
 	# assert the storage kind name
-	juju list-storage --format json | jq '.storage | .["disks/1"] | .kind' | check "block"
+	juju storage --format json | jq '.storage | .["disks/1"] | .kind' | check "block"
 	# assert the storage label name
-	juju list-storage --format json | jq '.storage| keys[0]' | check "disks/1"
+	juju storage --format json | jq '.storage| keys[0]' | check "disks/1"
 	# assert the unit attachment name
-	juju list-storage --format json | jq '.storage |.["disks/1"]|.attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
+	juju storage --format json | jq '.storage |.["disks/1"]|.attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
 	# assert the attached unit state
-	juju list-storage --format json | jq '.storage[] | .attachments |.units[] | .life' | check "alive"
+	juju storage --format json | jq '.storage[] | .attachments |.units[] | .life' | check "alive"
 	echo "Block loop disk 1 PASSED"
+
 	echo "Assessing add storage block loop disk 2"
 	juju add-storage -m "${model_name}" dummy-storage-lp/0 disks=1
 	# assert the storage kind name
-	juju list-storage --format json | jq '.storage| .["disks/2"]| .kind' | check "block"
+	juju storage --format json | jq '.storage| .["disks/2"]| .kind' | check "block"
 	# assert the storage label name
-	juju list-storage --format json | jq '.storage | keys[1]' | check "disks/2"
+	juju storage --format json | jq '.storage | keys[1]' | check "disks/2"
 	# assert the unit attachment name
-	juju list-storage --format json | jq '.storage | .["disks/2"] | .attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
+	juju storage --format json | jq '.storage | .["disks/2"] | .attachments |.units | keys | .[]' | check "dummy-storage-lp/0"
 	# assert the attached unit state
-	juju list-storage --format json | jq '.storage | .["disks/2"] | .life' | check "alive"
+	juju storage --format json | jq '.storage | .["disks/2"] | .life' | check "alive"
```

Comment:
Here you check the storage is alive, but everywhere else you check the life of the attached units `.attachments | .units[] | .life`